module/zfs/vdev_mirror.c

```diff
@@ -451,17 +452,17 @@ vdev_mirror_scrub_done(zio_t *zio)
 	mirror_child_t *mc = zio->io_private;
 
 	if (zio->io_error == 0) {
-		zio_t *pio;
-		zio_link_t *zl = NULL;
-
-		mutex_enter(&zio->io_lock);
-		while ((pio = zio_walk_parents(zio, &zl)) != NULL) {
-			mutex_enter(&pio->io_lock);
-			ASSERT3U(zio->io_size, >=, pio->io_size);
+		zio_t *pio = zio_unique_parent(zio);
```

Comment:
The original algorithm walked all parents to do the copy... while it seems reasonable that there should generally be only a single parent here, do you have a reason to believe that is an invariant? Why did the original algorithm assume the possibility of multiple parents?

Reply:
I have no idea what the original author was thinking about.  zio_unique_parent() actually verifies there are not other parents, and the assertion didn't fire on my tests.  My best guess is that it was some paranoia, same as unneeded locking.